tools/tsp-client/CHANGELOG.md

```diff
@@ -1,5 +1,9 @@
 # Release
 
+## 2025-04-16 - 0.21.0
+
+- Forward untracked dependencies in emitter-package.json when using the `generate-config-files` command.
```

Comment:
What do we mean by "untracked" dependencies? Are these dependencies that have been manually added to the emitter-package.json file? I thought we will always be generating the emitter-package.json file going forward with tsp-client.

Reply:
We've had several discussions around this and folks require this flexibility in certain cases. Allen has been working on a specific scenario where we want to enable this flexibility so manually pinned dependencies arent erased. Likewise the rust repo also wants to allow this for their repo. Do you have any specific concerns? Repo owners should still be reviewing any PRs updating their emitter-package.json file.

Reply:
I will update the wording here to say manually pinned so this change is clearer, thanks for pointing that out!

Reply:
I dont think we're there yet, but if in the future someone wants to reset their emitter-package.json to be purely based on the deps in the package.json of the target emitter, we can add a flag to force that

Reply:
I think the wording was my main feedback. I can think of scenarios where this would be useful for C#, e.g. for defining a service specific plugin to the emitter. That said, I think in that case we would end up duplicating the majority of the emitter-package.json between service and repo root, so I was planning on going another route to support the plugins.

Reply:
One other question - what if the actual package.json removes a devDependency? Won't the changes in this PR prevent that from ever getting removed in the emitter-package-json?

Reply:
Ah yeah I think in that case, we'd probably use the alternate path for emitter-package json, right?

Reply:
Tracking issue to add a flag to do a hard reset on emitter-package.json: https://github.com/Azure/azure-sdk-tools/issues/10333

Will add that in a separate PR, I dont think we need to block here on that.